File: Exemplos M10/App_IntentDigitalHUB_Ionic/Elgin_IntentDigitalHUB_Ionic-font/src/intentDigitalHubService/TERMICA/Commands/ImpressaoTexto.ts
import { TermicaCommand } from "../TermicaCommand";

export class ImpressaoTexto extends TermicaCommand {
  readonly dados: string;
  readonly posicao: number;
  readonly stilo: number;
  readonly tamanho: number;

  constructor(dados: string, posicao: number, stilo: number, tamanho: number) {
    super("ImpressaoTexto");
    this.dados = dados;
    this.posicao = posicao;
    this.stilo = stilo;
    this.tamanho = tamanho;
  }

  functionParameters(): object {
    return {
      dados: this.dados,
      posicao: this.posicao,
      stilo: this.stilo,
      tamanho: this.tamanho,
    };
  }
}

File: Exemplos SmartPOS/App_IntentDigitalHUB_Flutter/Elgin_IntentDigitalHUB_SmartPOS_Flutter-font/lib/IntentDigitalHubService/TERMICA/Commands/impressao_texto.dart
import '../termica_command.dart';

class ImpressaoTexto extends TermicaCommand {
  final String _dados;
  final int _posicao;
  final int _stilo;
  final int _tamanho;

  ImpressaoTexto(this._dados, this._posicao, this._stilo, this._tamanho)
      : super('ImpressaoTexto');

  @override
  get functionParametersJson => {
        'dados': _dados,
        'posicao': _posicao,
        'stilo': _stilo,
        'tamanho': _tamanho,
      };
}
